src/tcutility/analysis/vdd/manager.py

```diff
@@ -0,0 +1,220 @@
+        self.change_unit(unit)
+        frag_indices = [charge.frag_index for charge in self.vdd_charges["vdd"]]
+        atom_symbols = [f"{charge.atom_index}{charge.atom_symbol}" for charge in self.vdd_charges["vdd"]]
+        charges = [[charge.charge for charge in charges] for _, charges in self.vdd_charges.items()]
+        headers = ["Frag", "Atom"] + [irrep for irrep, _ in self.vdd_charges.items()]
+        combined_table = list(zip_longest(frag_indices, atom_symbols, *charges, fillvalue=""))
+        df = pd.DataFrame(combined_table, columns=headers).rename(columns={"vdd": "Total"})
+        return df
+
+    def get_summed_vdd_charges_dataframe(self, unit: str = "me") -> pd.DataFrame:
+        """Get the summed VDD charges as a pandas DataFrame in a specified unit ([me] or [e])."""
+        summed_data = self.get_summed_vdd_charges(unit)
+        summed_data["Frag"] = {str(key): int(key) for key in summed_data["vdd"].keys()}
+        df = pd.DataFrame(summed_data).pipe(lambda df: df[df.columns.tolist()[-1:] + df.columns.tolist()[:-1]])  # move the "Frag" column to the front
+        return df.rename(columns={"vdd": "Total"})
+
+    def get_vdd_charges_table(self, unit: str = "me") -> str:
+        df = self.get_vdd_charges_dataframe(unit)
+        return df.to_string(float_format=lambda x: PRINT_FORMAT[self.unit] % x, justify="center", index=False, col_space=6)
+
+    def get_summed_vdd_charges_table(self, unit: str = "me") -> str:
+        df = self.get_summed_vdd_charges_dataframe(unit)
+        return df.to_string(float_format=lambda x: PRINT_FORMAT[self.unit] % x, justify="center", col_space=6, index=False)
+
+    @staticmethod
+    def write_to_txt(output_dir: Union[str, pl.Path], managers: Union[VDDChargeManager, Sequence[VDDChargeManager]], unit: str = "me") -> None:
+        """Write the VDD charges to a text file. It is a static method because multiple managers can be written to the same file."""
+        out_dir = pl.Path(output_dir) if not isinstance(output_dir, pl.Path) else output_dir
+        files = [out_dir / "VDD_charges_per_atom.txt", out_dir / "VDD_charges_per_fragment.txt"]
+        managers = managers if isinstance(managers, Sequence) else [managers]
+
+        # Print charges per atom and per fragment in seperate files
+        for i, file in enumerate(files):
+            with open(file, "w") as f:
+                f.write(f"VDD charges (in unit {unit}):\n")
+                for manager in managers:
+                    f.write(f"{manager.name}\n")
+                    if i == 0:
+                        f.write(manager.get_vdd_charges_table(unit))
+                    else:
+                        f.write(manager.get_summed_vdd_charges_table(unit)) if manager.is_fragment_calculation else f.write("No fragment calculation\n")
+                    f.write("\n\n")
+
+    def write_to_excel(self, output_dir: Union[str, pl.Path], unit: str = "me") -> None:
```

Comment:
The unit specification is now gone indeed. The output_dir workings have changed for the plot and write_excel methods as you suggested. However, the write "write_to_text" method I won't change because it creates two files (charges per atom and per frag) with multiple managers. So I don't see the use of it printing the charges to each calculation directory; it should function as a summary method for more managers.

Reply:
I agree fully